pkg/v1/tkg/tkgconfigupdater/provider.go

```diff
@@ -46,17 +52,44 @@ func (c *client) defaultProviders() (providers, error) {
 		return providers{}, errors.Wrapf(err, "Unable to unmarshall provider config")
 	}
 	for i := range providersConfig.Providers {
-		path := filepath.Join(tkgDir, providersConfig.Providers[i].URL)
-		if strings.Contains(path, "\\") {
-			// convert windows backslash style paths 'c:\foo\....' to file:// urls
-			path = "file:///" + filepath.ToSlash(path)
-		}
-
-		providersConfig.Providers[i].URL = path
+		providersConfig.Providers[i].URL = getFilePath(tkgDir, providersConfig.Providers[i].URL)
 	}
+	providersConfig.CertManager.URL = getFilePath(tkgDir, providersConfig.CertManager.URL)
+
 	return providersConfig, nil
 }
 
+func getFilePath(tkgDir, url string) string {
+	path := filepath.Join(tkgDir, url)
+	if strings.Contains(path, "\\") {
+		// convert windows backslash style paths 'c:\foo\....' to file:// urls
+		path = "file:///" + filepath.ToSlash(path)
+	}
+	return path
+}
+
+func ensureCertManagerInConfig(defaultProviders providers, tkgConfigNode *yaml.Node) error {
+	certManagerIndex := GetNodeIndex(tkgConfigNode.Content[0].Content, constants.CertManagerConfigKey)
+	if certManagerIndex == -1 {
+		tkgConfigNode.Content[0].Content = append(tkgConfigNode.Content[0].Content, createSequenceNode(constants.CertManagerConfigKey)...)
+		certManagerIndex = GetNodeIndex(tkgConfigNode.Content[0].Content, constants.CertManagerConfigKey)
+	}
+
+	certManagerBytes, err := yaml.Marshal(defaultProviders.CertManager)
+	if err != nil {
+		return errors.Wrap(err, "unable to get cert-manager")
+	}
+
+	certManagerNode := yaml.Node{}
+	err = yaml.Unmarshal(certManagerBytes, &certManagerNode)
+	if err != nil {
+		return errors.Wrap(err, "unable to get cert-manager")
+	}
```

Comment:
suggestion: Extract as a helper func. It will simplify the code and improve readability. I've also noticed at least two other places where it could be used: see my comment on `EnsureProvidersInConfig()`.

The signature would look something like this:
```go
func copyData(from, to interface{}) error
```